includes/gateways/paypal/checkout-actions.php

```diff
@@ -154,32 +155,79 @@ function create_order( $purchase_data ) {
 			);
 		}
 
+		$order_subtotal = $purchase_data['subtotal'];
+		$items          = get_order_items( $purchase_data );
+		// Adjust the order subtotal if any items are discounted.
+		foreach ( $items as $item ) {
+			// A discount can be negative, so cast it to an absolute value for comparison.
+			if ( (float) abs( $item['discount'] ) > 0 ) {
+				$order_subtotal -= ( $item['discount'] * $item['quantity'] );
+			}
+
+			// The discount amount is not passed to PayPal as part of the $item.
+			unset( $item['discount'] );
+		}
+
+		$discount = 0;
+		// Fees which are not item specific need to be added to the PayPal data as order items.
+		if ( ! empty( $purchase_data['fees'] ) ) {
+			foreach ( $purchase_data['fees'] as $fee ) {
+				if ( ! empty( $fee['download_id'] ) ) {
+					continue;
+				}
+				// Positive fees.
+				if ( floatval( $fee['amount'] ) > 0 ) {
+					$items[]         = array(
+						'name'        => stripslashes_deep( html_entity_decode( wp_strip_all_tags( $fee['label'] ), ENT_COMPAT, 'UTF-8' ) ),
+						'unit_amount' => array(
+							'currency_code' => $currency,
+							'value'         => edd_sanitize_amount( $fee['amount'] ),
+						),
+						'quantity'    => 1,
+					);
+					$order_subtotal += abs( $fee['amount'] );
+				} else {
+					// This is a negative fee (discount) not assigned to a specific Download
+					$discount += abs( $fee['amount'] );
+				}
+			}
+		}
+
+		$tax          = (float) $purchase_data['tax'] > 0 ? $purchase_data['tax'] : 0;
 		$order_amount = array(
-			'currency_code' => edd_get_currency(),
-			'value'         => (string) $purchase_data['price']
-		);
-		if ( (float) $purchase_data['tax'] > 0 ) {
-			$order_amount['breakdown'] = array(
+			'currency_code' => $currency,
+			'value'         => (string) ( $order_subtotal + $tax - $discount ),
+			'breakdown'     => array(
```

Comment:
I'm feeling semi uncomfortable about the way these breakdowns work. We're essentially recalculating the subtotal, discount, tax, and order total, instead of using the values provided in `$purchase_data`. My concern being ending up in another situation like this: https://github.com/easydigitaldownloads/easy-digital-downloads/issues/6837

If we ever change order calculations somehow, we now have to update that logic in two places (standard purchase data place, and PayPal-specific overview).

One of the things I do want to do at some point is improve the information that's currently available in `$purchase_data`, because it could definitely be more useful, but I guess my point still stands that I'm still feeling concerned about how much we're rewriting here.

Is there any reason we can't just use the values already available in `$purchase_data` ?

Reply:
I think maybe we can for the overall order value. Looking back through my commits, it looks like I changed that because the item_total value was giving me fits and I was mentally tying them together. I think the item_total one does need to be handled differently because of how taxes, discounts, and fees end up resolving, but some tentative tests suggest that this can be reverted to `$purchase_data['price']`.